data/data/openstack/bootstrap/main.tf

```diff
@@ -0,0 +1,46 @@
+resource "openstack_objectstorage_object_v1" "ignition" {
+  container_name = "${var.swift_container}"
+  name           = "bootstrap.ign"
+  content        = "${var.ignition}"
+}
+
+resource "openstack_objectstorage_tempurl_v1" "ignition_tmpurl" {
+  container = "${var.swift_container}"
+  method    = "get"
+  object    = "${openstack_objectstorage_object_v1.ignition.name}"
+  ttl       = 3600
+}
+
+data "ignition_config" "redirect" {
+  replace {
+    source = "${openstack_objectstorage_tempurl_v1.ignition_tmpurl.url}"
+  }
+}
+
+data "openstack_images_image_v2" "bootstrap_image" {
+  name        = "${var.image_name}"
+  most_recent = true
+}
+
+data "openstack_compute_flavor_v2" "bootstrap_flavor" {
+  name = "${var.flavor_name}"
+}
+
+resource "openstack_compute_instance_v2" "bootstrap" {
+  name      = "${var.cluster_name}-bootstrap"
+  flavor_id = "${data.openstack_compute_flavor_v2.bootstrap_flavor.id}"
+  image_id  = "${data.openstack_images_image_v2.bootstrap_image.id}"
+
+  user_data = "${data.ignition_config.redirect.rendered}"
```

Comment:
Is the bootstrap Ignition config too large to be directly assigned? I'm not sure if OpenStack has limits on the size of userdata.

Reply:
Yes, I'm pretty sure that's why it was done this way.  Google suggest a 64kB limit.

Reply:
Yes that's correct, we hit the size limit in early testing so moved to a swift URL instead of directly providing the user-data.  Perhaps in future we might consider a config drive as an alternative, or figure out a way to make the ignition config smaller.